src/session.rs

```diff
@@ -10,24 +10,48 @@ use std::pin::Pin;
 use std::task::Context;
 use std::task::Poll;
 use tokio::sync::{mpsc, oneshot};
-use webdriver::command::WebDriverCommand;
+use webdriver::command::{WebDriverCommand, WebDriverExtensionCommand, VoidWebDriverExtensionCommand};
 use webdriver::error::ErrorStatus;
 use webdriver::error::WebDriverError;
 
 type Ack = oneshot::Sender<Result<Json, error::CmdError>>;
 
 /// A WebDriver client tied to a single browser session.
 #[derive(Clone, Debug)]
-pub struct Client {
-    tx: mpsc::UnboundedSender<Task>,
+pub struct Client<T: ExtensionCommand> {
```

Comment:
Making the `Client` generic over this is a _huge_ and backwards-incompatible change that I'd prefer to avoid. I think it'd be better to instead use a trait object, and pass around `Box<dyn WebDriverExtensionCommand>`. That way you can get rid of all these `<T>` :)

Reply:
@jonhoo I am getting an error when using `Box<dyn WebDriverExtensionCommand>`.  Error:- `the trait `WebDriverExtensionCommand` cannot be made into an object` . Because `WebDriverExtensionCommand` is extend to `PartialEq`. Can you help me to resolve this:- https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=91be6634f09c530866951321286b8d2c

Reply:
Oh, hmm, that's really awkward... That seems like a bug in the `WebDriverExtensionCommand` trait to me, which will unfortunately require reporting a bug to that crate to resolve. Without that, there isn't a way for us to add this functionality without significantly complicating the API for (seemingly) no reason.

Reply:
@whizsid did you end up filing an issue with the upstream crate for this?

Reply:
@jonhoo Yes. I submitted an issue. They shipped it with the latest release. I will update my PR.